packages/nextjs/src/config/webpack.ts

```diff
@@ -94,9 +94,9 @@ export function constructWebpackConfigFunction(
         newConfig.module.rules.push({
           // Nextjs allows the `pages` folder to optionally live inside a `src` folder
           test: new RegExp(
-            `${escapeStringForRegex(projectDir)}${
+            `^${escapeStringForRegex(projectDir)}${
               shouldIncludeSrcDirectory ? '/src' : ''
-            }/pages/.*\\.(${pageExtensionRegex})`,
+            }/pages/.*\\.(${pageExtensionRegex})$`,
```

Comment:
I have to admit this feels a bit risky because of file path shenanigans (windows paths?). I am open to other suggestions.

Making it an exact match should be fine in combination with our query-string import approach because we don't want to process these files anyways. Maybe we can even remove the exit condition (`if (this.resourceQuery.includes('__sentry_wrapped__')) return userCode;`)

Reply:
We should have two regex - one for windows and one for Unix if this one does not support windows imo

Reply:
Ugh, right, I always forget about windows path separators. Good call, @AbhiPrasad.

I don't think we need to do it in two regexes, though. What if we did something like [this](https://github.com/getsentry/sentry-javascript/compare/master...temp-10-24-page-regex)?

![image](https://user-images.githubusercontent.com/14812505/197663659-d0eafe06-8198-48ef-a2c5-8d3102a1cd1e.png)

(I didn't use `pageExtensionsRegex` because `escapeStringForRegex` isn't idempotent.)

Reply:
Ah perfect, let’s do that then!

Reply:
Katie that change gets rid of the solution for the `src` folder problematic: https://github.com/getsentry/sentry-javascript/issues/5978

I like however that we build the path to the Next.js `pages` before we pass it to `escapeStringForRegex()` so I opted for a hybrid approach. [43f2823](https://github.com/getsentry/sentry-javascript/pull/6013/commits/43f2823b8a2011d07d2a45b06135852e426dc7e8)

Reply:
But we shouldn't include both! It's either `pages` or `src/pages`.

Reply:
No, I mean: If a user has their pages in `/pages`, then that will be reflected in the value of `pagesDir`. If a user has their pages in `src/pages`, that will also be reflected in the value of `pagesDir.` Either way, we don't need to recalculate the value.

If I do

```js
console.log(pagesDir);
console.log(pagesDirectory);
```

depending on whether or not I'm using a `src` directory I get either

```
/Users/Katie/Documents/Sentry/test-apps/next-js-test/src/pages
/Users/Katie/Documents/Sentry/test-apps/next-js-test/src/pages
```

or

```
/Users/Katie/Documents/Sentry/test-apps/next-js-test/pages
/Users/Katie/Documents/Sentry/test-apps/next-js-test/pages
```

Reply:
Oh sorry got confused and didn't realize that `pagesDir` was a thing. I'll approve that PR!